plugin.video.montreal.greek-tv/resources/lib/default.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+import urllib.parse, os, sys
+import xbmcvfs, xbmcgui, xbmcplugin, xbmcaddon
+
+addon_id       = 'plugin.video.montreal.greek-tv'
+Home           = xbmcvfs.translatePath(xbmcaddon.Addon().getAddonInfo('path'))
+art            = xbmcvfs.translatePath(Home + '/resources/art/')
+icon           = xbmcvfs.translatePath(os.path.join(Home, 'icon.png'))
+tv_icon        = xbmcvfs.translatePath(os.path.join(art , 'montreal_tv.png'))
+radio_icon     = xbmcvfs.translatePath(os.path.join(art , 'montreal_radio.png'))
+radio_fanart   = xbmcvfs.translatePath(os.path.join(art , 'montreal_backround.jpg'))
+
+__handle__     = int(sys.argv[1])
+
+TV_URL = 'http://live.greektv.ca/hls1/greektv.m3u8'
+RADIO_URL = 'http://live.greekradio.ca:8000/live'
+
+def Index():
+    addDir("Montreal Greek TV", TV_URL, 2, tv_icon, radio_fanart)
+    addDir("Montreal Greek Radio", RADIO_URL, 2, radio_icon, radio_fanart)
+    xbmcplugin.endOfDirectory(__handle__)
+
+def addDir(name, url, mode, iconimage, fanart):
+        u = sys.argv[0]+"?url="+urllib.parse.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.parse.quote_plus(name)+"&iconimage="+urllib.parse.quote_plus(iconimage)+"&fanart="+urllib.parse.quote_plus(fanart)
+        liz = xbmcgui.ListItem(name)
+        liz.setArt({'thumb': iconimage, "fanart": fanart})
+        liz.setInfo(type = "Video", infoLabels = {"Title": name, "Plot": "" } )
```

Comment:
SetInfo is deprecated in v20, probably a good idea to look into the new added classes for this (not mandatory)

Reply:
> SetInfo is deprecated in v20, probably a good idea to look into the new added classes for this (not mandatory)

I have try without setInfo but i get message when play with v19 matrix:

> 2022-08-29 15:31:24.209 T:11108 WARNING <general>: Playlist Player: ListItem type must be audio or video, use ListItem::setInfo to specify!

and replace with InfoTagVideo not clear for me:
@python_v20 Partially deprecated. Use explicit setters in InfoTagVideo, InfoTagMusic, InfoTagPicture or InfoTagGame instead.

Reply:
Yeah the change is not mandatory as it is still supported, just a heads up on the deprecation from v20 and up

Reply:
ok i will look that for v20 version. Thank you